# caribou_MiSeq_project
Repository for maintaining scripts and tools used in processing and analysis of caribou (Rangifer tarandus) high-throughput sequencing amplicon data. 


## Workflows

The scripts herein contribute to 5 unique workflows, each with the purpose of caribou genotyping


### Microsatellite Profile Genotyping

1. trim.sh
2. megasat.sh


### Sex Identification

1. trim.sh
2. sexID.sh


### PRNP Disease Tolerance Genotyping

1. trim.sh
2. prnp_align.sh
3. prnp_vcf.sh


### DRB Immunity Haplotype Genotyping (IN PROGRESS)

1. snp_trim.sh
2. drb_align.sh
3. drb_vcf.sh


### Genome Alignment

1. trim.sh
2. genome_align.sh




## Process Descriptions


### trim.sh

Used for quality-trimming and quality assessment. **Trimmomatic**[^1] removes adapter sequences, removes low-quality sequences, and removes truncated sequences.  **Fastqc**[^2] generates quality reports for trimmed sequence data. **Multiqc**[^3] combines reports generated by Fastqc to easily summarize quality information.


### snp_trim.sh

Insert text here


### genome_align.sh

**Bowtie2**[^4] aligns trimmed sequence reads to the caribou reference genome. **Samtools**[^5] takes SAM output from Bowtie2 and converts it to a set of sorted, compressed BAM files.


### megasat.sh

megasat.sh runs **MEGASAT_Genotype.pl**[^6], taking the output from trim.sh as input and generating a file of microsatellite genotypes.


### sexID.sh

**Bowtie2** aligns trimmed sequence reads to the caribou ZFX/Y locus for sex identification. Bowtie2 SAM output is converted to BAM using **samtools*, then a custom script generates a sexID table for all samples run.


### prnp_align.sh

Insert text here


### prnp_vcf.sh

Insert text here


### drb_align.sh

Insert text here


### drb_vcf.sh

Insert text here




[^1]: Bolger, A. M., Lohse, M., & Usadel, B. (2014). Trimmomatic: A flexible trimmer for Illumina sequence data. Bioinformatics, 30(15), 2114–2120. [https://doi.org/10.1093/bioinformatics/btu170]
[^2]: https://www.bioinformatics.babraham.ac.uk/projects/fastqc/
[^3]: Ewels, P., Magnusson, M., Lundin, S., & Käller, M. (2016). MultiQC: Summarize analysis results for multiple tools and samples in a single report. Bioinformatics, 32(19), 3047–3048. [https://doi.org/10.1093/bioinformatics/btw354]
[^4]: Langmead, B., Trapnell, C., Pop, M., & Salzberg, S. L. (2009). Ultrafast and memory-efficient alignment of short DNA sequences to the human genome. Genome Biology, 10(3). [https://doi.org/10.1186/gb-2009-10-3-r25]
[^5]: Danecek, P., Bonfield, J. K., Liddle, J., Marshall, J., Ohan, V., Pollard, M. O., … Li, H. (2021). Twelve years of SAMtools and BCFtools. GigaScience, 10(2), 1–4. [https://doi.org/10.1093/gigascience/giab008]
[^6]: Zhan, L., Paterson, I. G., Fraser, B. A., Watson, B., Bradbury, I. R., Nadukkalam Ravindran, P., … Bentzen, P. (2017). MEGASAT: Automated Inference of Microsatellite Genotypes From Sequence Data. Molecular Ecology Resources, 17(2), 247–256. [https://doi.org/10.1111/1755-0998.12561]
